src/devices/Card/Icode/IcodeCard.cs

```diff
@@ -0,0 +1,360 @@
+        /// </summary>
+        /// <param name="rfid">A card transceiver class</param>
+        /// <param name="target">The target number as some card readers attribute one</param>
+        public IcodeCard(CardTransceiver rfid, byte target)
+        {
+            _rfid = rfid;
+            Target = target;
+            _logger = this.GetCurrentClassLogger();
+        }
+
+        /// <summary>
+        /// Provide a calculation of CRC for ISO15693
+        /// </summary>
+        /// <param name="buffer">The buffer to process</param>
+        /// <param name="crc">The CRC, Must be a 2 bytes buffer</param>
+        public void CalculateCrc15693(ReadOnlySpan<byte> buffer, Span<byte> crc)
+        {
+            if (crc.Length != 2)
+            {
+                throw new ArgumentException($"Value must be 2 bytes.", nameof(crc));
+            }
+
+            ushort polynomial = 0x8408;
+            ushort currentCrc = 0xFFFF;
+            // ISO15693-3.pdf
+            for (int i = 0; i < buffer.Length; i++)
+            {
+                currentCrc = (ushort)(currentCrc ^ buffer[i]);
+                for (int j = 0; j < 8; j++)
+                {
+                    if ((currentCrc & 0x0001) != 0)
+                    {
+                        currentCrc = (ushort)((currentCrc >> 1) ^ polynomial);
+                    }
+                    else
+                    {
+                        currentCrc = (ushort)(currentCrc >> 1);
+                    }
+                }
+            }
+
+            currentCrc = (ushort)~currentCrc;
+            crc[0] = (byte)(currentCrc & 0xFF);
+            crc[1] = (byte)((currentCrc >> 8) & 0xFF);
+        }
+
+        /// <summary>
+        /// Run the last setup command. In case of reading bytes, they are automatically pushed into the Data property
+        /// </summary>
+        /// <returns>-1 if the process fails otherwise the number of bytes read</returns>
+        public int RunIcodeCardCommand()
+        {
+            byte[] requestData = Serialize();
+            byte[] dataOut = new byte[_responseSize];
+
+            var ret = _rfid.Transceive(Target, requestData, dataOut.AsSpan(), NfcProtocol.Iso15693);
+            _logger.LogDebug($"{nameof(RunIcodeCardCommand)}: {Command}, Target: {Target}, Data: {BitConverter.ToString(Serialize())}, Success: {ret}, Dataout: {BitConverter.ToString(dataOut)}");
+            if (ret > 0)
+            {
+                Data = dataOut;
+            }
+
+            return ret;
+        }
+
+        /// <summary>
+        /// Depending on the command, serialize the needed data
+        /// Reading data will just serialize the command
+        /// Writing data will serialize the data as well
+        /// </summary>
+        /// <returns>The serialized bits</returns>
+        private byte[] Serialize()
+        {
+            byte[]? ser = null;
+            // Span<byte> crc = new Span<byte>(new byte[] { 0xff, 0xff });
```

Comment:
why do you have dead code commented?
If there is a reason, please add a comment otherwise, please remove all the dead code from this function.